<!DOCTYPE html>
<!-- saved from url=(0043)https://sourceforge.net/p/mageck/wiki/Home/ -->
<html lang="en" class="js flexbox canvas canvastext no-touch geolocation postmessage websqldatabase indexeddb hashchange history draganddrop websockets rgba hsla multiplebgs backgroundsize borderimage borderradius boxshadow textshadow opacity cssanimations csscolumns cssgradients cssreflections csstransforms csstransforms3d csstransitions fontface video audio localstorage sessionstorage webworkers applicationcache svg inlinesvg smil svgclippaths js canvas canvastext"><!--<![endif]--><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    
    <title>MAGeCK / Wiki / Home</title>
    
<meta id="webtracker" name="webtracker" content="{&quot;event_id&quot;: &quot;c6139eac-ae0f-11e4-8173-0200ac1d2942&quot;, &quot;project&quot;: &quot;mageck&quot;, &quot;user_id&quot;: 4568329, &quot;action_type&quot;: &quot;wiki&quot;}">

<script type="text/javascript" async="" src="./MAGeCK_Wiki_Home_files/ga.js"></script><script src="./MAGeCK_Wiki_Home_files/modernizr.custom.90514.js"></script>

<script src="./MAGeCK_Wiki_Home_files/jquery-1.8.0.min.js"></script>

<script src="./MAGeCK_Wiki_Home_files/header.js"></script>
<!--[if lt IE 7 ]>
  <script src="https://a.fsdn.com/allura/nf/1422305739/_ew_/theme/sftheme/js/sftheme/dd_belatedpng.js"></script>
  <script> DD_belatedPNG.fix('img, .png_bg'); //fix any <img> or .png_bg background-images </script>
<![endif]-->
<link href="./MAGeCK_Wiki_Home_files/css" rel="stylesheet" type="text/css">
<style type="text/css">
    @font-face {
        font-family: "Pictos";
        src: url('https://a.fsdn.com/allura/nf/1422305739/_ew_/theme/sftheme/css/fonts/sftheme/pictos-web.eot');
        src: local("â˜º"), url('https://a.fsdn.com/allura/nf/1422305739/_ew_/theme/sftheme/css/fonts/sftheme/pictos-web.woff') format('woff'), url('https://a.fsdn.com/allura/nf/1422305739/_ew_/theme/sftheme/css/fonts/sftheme/pictos-web.ttf') format('truetype'), url('https://a.fsdn.com/allura/nf/1422305739/_ew_/theme/sftheme/css/fonts/sftheme/pictos-web.svg') format('svg');
    }
</style><style type="text/css"></style>
    <script type="text/javascript">
            /*jslint onevar: false, nomen: false, evil: true, css: true, plusplus: false, white: false, forin: true, on: true, immed: false */
            /*global confirm, alert, unescape, window, jQuery, $, net, COMSCORE */
    </script>
    
      <!-- ew:head_css -->

    
      <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/hilite.css">
    
      <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/wiki.css">
    
      <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/tool_icon_css">
    
      <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/forge.css">
    
      <link rel="stylesheet" type="text/css" href="./MAGeCK_Wiki_Home_files/css(1)">
    
      
<!-- /ew:head_css -->

    
    
    
      <!-- ew:head_js -->

    
      
<!-- /ew:head_js -->

    
    

    
      <style type="text/css">
        #page-body.project---init-- #top_nav { display: none; }
#page-body.project---init-- #nav_menu_holder { display: none; margin-bottom: 0; }
#page-body.project---init-- #content_base {margin-top: 0; }
      </style>
    
    
<style>
  .hidden { display: none }
  a.notfound { color: #f00; }
</style>

      <style>.XRnlGCMfAwvhuIKzzFoUtM { display:none }</style>

    
<link rel="alternate" type="application/rss+xml" title="Page RSS" href="https://sourceforge.net/p/mageck/wiki/Home/feed.rss">
<link rel="alternate" type="application/atom+xml" title="Page Atom" href="https://sourceforge.net/p/mageck/wiki/Home/feed.atom">
<link rel="alternate" type="application/rss+xml" title="Wiki RSS" href="https://sourceforge.net/p/mageck/wiki/feed.rss">
<link rel="alternate" type="application/atom+xml" title="Wiki Atom" href="https://sourceforge.net/p/mageck/wiki/feed.atom">

    
    


<script type="text/javascript">
    var _gaq = _gaq || [];

    function _add_tracking(prefix, tracking_id, send_user) {
        _gaq.push([prefix+'._setAccount', tracking_id]);
        _gaq.push([prefix+'._setCustomVar', 1, 'Page Type', 'wiki', 3]);
        if (send_user) {
            _gaq.push([prefix+'._setCustomVar', 2, 'userid', '4568329', 1]);
        }_gaq.push([prefix+'._trackPageview']);
    }
      _add_tracking('sfnt1', 'UA-32013-6', true);
      _add_tracking('sfnt2', 'UA-36130941-1', true);
    

    (function() {
        var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
        ga.src = ('https:' === document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
</script>
  <style type="text/css">/**
 * Style sheet used by new LibX tooltip code
 */

/* We insert a <div> with libx-tooltip style under the body.
 * This will inherit body's style - we can't afford to inherit undesirable 
 * styles and we must redefine what we need.  OTOH, some things, e.g.
 * font-size, might be ok to be inherited to stay within the page's tone.
 */
.libx-tooltip {
    display: none;
    overflow: visible;
    padding: 5px;
    z-index: 100;
    background-color: #eee;
    color: #000;
    font-weight: normal;
    font-style: normal;
    text-align: left;
    border: 2px solid #666;
    border-radius: 5px;
    -webkit-border-radius: 5px;
    -moz-border-radius: 5px;
}

.libx-tooltip p {
    /* override default 1em margin to keep paragraphs inside a tooltip closer together. */
    margin: .2em;
}
</style><style type="text/css">/**
 * Style sheet used by LibX autolinking code
 */
.libx-autolink {
}

</style><style type="text/css">div.jGrowl {
	padding: 			10px;
	z-index: 			9999;
}

/** Special IE6 Style Positioning **/
div.ie6 {
	position: 			absolute;
}

div.ie6.top-right {
	right: 				auto;
	bottom: 			auto;
	left: 				expression( ( 0 - jGrowl.offsetWidth + ( document.documentElement.clientWidth ? document.documentElement.clientWidth : document.body.clientWidth ) + ( ignoreMe2 = document.documentElement.scrollLeft ? document.documentElement.scrollLeft : document.body.scrollLeft ) ) + 'px' );
  	top: 				expression( ( 0 + ( ignoreMe = document.documentElement.scrollTop ? document.documentElement.scrollTop : document.body.scrollTop ) ) + 'px' );
}

div.ie6.top-left {
	left: 				expression( ( 0 + ( ignoreMe2 = document.documentElement.scrollLeft ? document.documentElement.scrollLeft : document.body.scrollLeft ) ) + 'px' );
	top: 				expression( ( 0 + ( ignoreMe = document.documentElement.scrollTop ? document.documentElement.scrollTop : document.body.scrollTop ) ) + 'px' );
}

div.ie6.bottom-right {
	left: 				expression( ( 0 - jGrowl.offsetWidth + ( document.documentElement.clientWidth ? document.documentElement.clientWidth : document.body.clientWidth ) + ( ignoreMe2 = document.documentElement.scrollLeft ? document.documentElement.scrollLeft : document.body.scrollLeft ) ) + 'px' );
	top: 				expression( ( 0 - jGrowl.offsetHeight + ( document.documentElement.clientHeight ? document.documentElement.clientHeight : document.body.clientHeight ) + ( ignoreMe = document.documentElement.scrollTop ? document.documentElement.scrollTop : document.body.scrollTop ) ) + 'px' );
}

div.ie6.bottom-left {
	left: 				expression( ( 0 + ( ignoreMe2 = document.documentElement.scrollLeft ? document.documentElement.scrollLeft : document.body.scrollLeft ) ) + 'px' );
	top: 				expression( ( 0 - jGrowl.offsetHeight + ( document.documentElement.clientHeight ? document.documentElement.clientHeight : document.body.clientHeight ) + ( ignoreMe = document.documentElement.scrollTop ? document.documentElement.scrollTop : document.body.scrollTop ) ) + 'px' );
}

div.ie6.center {
	left: 				expression( ( 0 + ( ignoreMe2 = document.documentElement.scrollLeft ? document.documentElement.scrollLeft : document.body.scrollLeft ) ) + 'px' );
	top: 				expression( ( 0 + ( ignoreMe = document.documentElement.scrollTop ? document.documentElement.scrollTop : document.body.scrollTop ) ) + 'px' );
	width: 				100%;
}

/** Normal Style Positions **/
body > div.jGrowl {
	position:			fixed;
}

body > div.jGrowl.top-left {
	left: 				0px;
	top: 				0px;
}

body > div.jGrowl.top-right {
	right: 				0px;
	top: 				0px;
}

body > div.jGrowl.bottom-left {
	left: 				0px;
	bottom:				0px;
}

body > div.jGrowl.bottom-right {
	right: 				0px;
	bottom: 			0px;
}

body > div.jGrowl.center {
	top: 				0px;
	width: 				50%;
	left: 				25%;
}

/** Cross Browser Styling **/
div.center div.jGrowl-notification, div.center div.jGrowl-closer {
	margin-left: 		auto;
	margin-right: 		auto;
}

div.jGrowl div.jGrowl-notification, div.jGrowl div.jGrowl-closer {
	background-color: 		#ccc;
	color: 					#000;
	opacity: 				.95;
	filter: 				alpha(opacity = 95);
	zoom: 					1;
	/* width: 					235px; */
	padding: 				10px;
	margin-top: 			5px;
	margin-bottom: 			5px;
	font-family: 			Tahoma, Arial, Helvetica, sans-serif;
	font-size: 				12px;
	text-align: 			left;
	display: 				none;
	-moz-border-radius: 	5px;
	-webkit-border-radius:	5px;
}

div.jGrowl div.jGrowl-notification {
	min-height: 			40px;
}

div.jGrowl div.jGrowl-notification div.header {
	font-weight: 			bold;
	font-size:				10px;
}

div.jGrowl div.jGrowl-notification div.close {
	float: 					right;
	font-weight: 			bold;
	font-size: 				12px;
	cursor:					pointer;
}

div.jGrowl div.jGrowl-closer {
	height: 				15px;
	padding-top: 			4px;
	padding-bottom: 		4px;
	cursor: 				pointer;
	font-size:				11px;
	font-weight: 			bold;
	text-align: 			center;
}
</style></head>

  <body class="wiki-Home" id="forge">
    <h2 class="hidden">
        <span style="color:red">Error:</span> CSS did not load.<br>
        This may happen on the first request due to CSS mimetype issues.
        Try clearing your browser cache and refreshing.
        <hr>
    </h2>
    
    
      <!-- ew:body_top_js -->

    
      
<!-- /ew:body_top_js -->

    
    
    
<header id="site-header">
    <div class="wrapper">
        <a href="https://sourceforge.net/" class="logo">
            <span>SourceForge</span>
        </a>
        
        <form method="get" action="https://sourceforge.net/directory/">
            <input type="text" id="words" name="q" placeholder="Search" tabindex="1">
        </form>
        
        <!--Switch to {language}-->
        <nav id="nav-site">
            <a href="https://sourceforge.net/directory/" title="Browse our software.">Browse</a>
            <a href="https://sourceforge.net/directory/enterprise" title="Browse our Enterprise software.">Enterprise</a>
            <a href="https://sourceforge.net/blog/" title="Read the latest news from the SF HQ.">Blog</a>
            <a href="https://sourceforge.net/support" title="Contact us for help and feedback.">Help</a>
            <a href="https://sourceforge.net/jobs?source=header" title="Search 80k+ tech jobs." class="featured-link">Jobs</a>
        </nav>
        <nav id="nav-account">
            
            <div id="account-tooltip" class="tooltip-container">
                <a href="https://sourceforge.net/users/davidliwei">
                    
  
    <img src="./MAGeCK_Wiki_Home_files/b9cc1f9048ed9e437d36915feecd38c5" alt="Wei Li" title="Wei Li" class="emboss x20">
  
                    Me â–¾
                </a>
                <div class="tooltip account">
                    <div class="diamond"></div>
                    <ul>
                        <li>
                            
  
    <img src="./MAGeCK_Wiki_Home_files/b9cc1f9048ed9e437d36915feecd38c5" alt="Wei Li" title="Wei Li" class="emboss x20">
  
                            <a href="https://sourceforge.net/users/davidliwei">Profile</a>
                        </li>
                        <li>
                            <b data-icon="U" class="ico ico-user" title="Account"></b>
                            <a href="https://sourceforge.net/auth/preferences/">Account Settings</a>
                        </li>
                        <li>
                            <b data-icon="Q" class="ico ico-power" title="Log Out"></b>
                            <a href="https://sourceforge.net/auth/logout">Log Out</a>
                        </li>
                    </ul>
                </div>
            </div>
             <!-- Change this when update notifications are available in Allura -->
            <div id="updater-tooltip" class="tooltip-container">
                
  
<a href="https://sourceforge.net/user/updates" title="Manage Update Notifications" class=" updates ">5</a>
<div class="tooltip updater">
    <h6>Available Updates</h6>
    <div class="diamond"></div>
    <ul>
        <li>The following software titles have updates:</li>
    
        <li>
            
            <a href="https://sourceforge.net/projects/matplotlib/?source=updater"><img alt="Icon" src="./MAGeCK_Wiki_Home_files/icon"></a>
            <a title="matplotlib" href="https://sourceforge.net/projects/matplotlib/?source=updater">matplotlib</a>
            <a href="https://sourceforge.net/projects/matplotlib/files/latest/download?source=updater" class="ico ico-downarrow" data-icon="}"></a>
        </li>
    
        <li>
            
            <a href="https://sourceforge.net/projects/mageck/?source=updater"><img alt="Icon" src="./MAGeCK_Wiki_Home_files/project_default.png"></a>
            <a title="MAGeCK" href="https://sourceforge.net/projects/mageck/?source=updater">MAGeCK</a>
            <a href="https://sourceforge.net/projects/mageck/files/latest/download?source=updater" class="ico ico-downarrow" data-icon="}"></a>
        </li>
    
        <li>
            
            <a href="https://sourceforge.net/projects/keepass/?source=updater"><img alt="Icon" src="./MAGeCK_Wiki_Home_files/icon(1)"></a>
            <a title="KeePass Password Safe" href="https://sourceforge.net/projects/keepass/?source=updater">KeePass Password ...</a>
            <a href="https://sourceforge.net/projects/keepass/files/latest/download?source=updater" class="ico ico-downarrow" data-icon="}"></a>
        </li>
    
        <li>
            
            <a href="https://sourceforge.net/projects/seqgene/?source=updater"><img alt="Icon" src="./MAGeCK_Wiki_Home_files/project_default.png"></a>
            <a title="seqgene" href="https://sourceforge.net/projects/seqgene/?source=updater">seqgene</a>
            <a href="https://sourceforge.net/projects/seqgene/files/latest/download?source=updater" class="ico ico-downarrow" data-icon="}"></a>
        </li>
    
        <li>
            
            <a href="https://sourceforge.net/projects/jtreeview/?source=updater"><img alt="Icon" src="./MAGeCK_Wiki_Home_files/project_default.png"></a>
            <a title="Java Treeview" href="https://sourceforge.net/projects/jtreeview/?source=updater">Java Treeview</a>
            <a href="https://sourceforge.net/projects/jtreeview/files/latest/download?source=updater" class="ico ico-downarrow" data-icon="}"></a>
        </li>
    
        <li><a href="https://sourceforge.net/user/updates">Manage Subscriptions</a></li>
    </ul>
</div>
            </div>
            
        </nav>
        
    </div>
</header>
<header id="site-sec-header">
    <div class="wrapper">
        <nav id="nav-hubs">
            <h4>Solution Centers</h4>
            <a href="http://goparallel.sourceforge.net/">Go Parallel</a>
            <a href="http://ibmsmarteritservices.sourceforge.net/">Smarter IT</a>
            <a href="http://ibmsoftwaredelivery.sourceforge.net/">Software Delivery</a>
            <a href="http://apmsolutions.sourceforge.net/">Performance</a>
            <a href="http://ibmsmarterdata.sourceforge.net/">Data Management</a>
        </nav>
        <nav id="nav-collateral">
            <a href="http://library.slashdotmedia.com/?source=sfnet_header">Resources</a>
            
        </nav>
    </div>
</header>
    
    
    
    
    
    <section id="page-body" class=" neighborhood-Projects project-mageck mountpoint-wiki">
	  <div id="nav_menu_holder">
            
            



    
    
    
    
    <nav id="breadcrumbs">
        <ul>
            <li itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb"><a itemprop="url" href="https://sourceforge.net/">Home</a></li>
            <li itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb"><a itemprop="url" href="https://sourceforge.net/directory">Browse</a></li>
            
            
                
            
            
            
                <li itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb"><a itemprop="url" href="https://sourceforge.net/p/">Projects</a></li>
                
            
            
                <li itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb"><a itemprop="url" href="https://sourceforge.net/p/mageck/">MAGeCK</a></li>
                
            
            
                <li itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb">Wiki</li>
                
            
        </ul>
    </nav>
    
    
  
      
    
    <div id="LeadGen_Developer_page" class="ad">
    <iframe height="0" width="0" src="about:blank" seamless="" frameborder="0" scrolling="no" style="display: none !important; visibility: hidden !important; opacity: 0 !important; background-position: 384px 70px;"></iframe>
    <noscript>&lt;a href="//ad.doubleclick.net/N4140/jump/ostg.sourceforge/LeadGen_Developer_page;sz=384x70;logged_in=1;tile=1;u=e%3Ac6139eac-ae0f-11e4-8173-0200ac1d2942;ord=3249013667480440?"&gt;&lt;img src="//ad.doubleclick.net/N4140/ad/ostg.sourceforge/LeadGen_Developer_page;sz=384x70;logged_in=1;tile=1;u=e%3Ac6139eac-ae0f-11e4-8173-0200ac1d2942;ord=3249013667480440?" width="384" height="70" /&gt;&lt;/a&gt;</noscript>
    </div>
  
    
    <h1 class="project_title">
        <a href="https://sourceforge.net/p/mageck/" class="project_link">MAGeCK</a>
    </h1>
    
    
    
    <span id="dev-status" class="prealpha">prealpha</span>
    
    <h2 class="project_summary">
        Model-based Analysis of Genome-wide CRISPR-Cas9 Knockout 
    </h2>
    
    <div class="brought-by">
        Brought to you by:
        
        
            
                <a href="https://sourceforge.net/u/davidliwei/">davidliwei</a>
            </div>
    

            
      </div>
      <div id="top_nav" class="">
        
        
<ul class="dropdown">
  
    <li class="">
        <a href="https://sourceforge.net/projects/mageck/" class="ui-icon-tool-summary-32">
            Summary
        </a>
        
        
    </li>
	
    <li class="">
        <a href="https://sourceforge.net/projects/mageck/files/" class="ui-icon-tool-files-32">
            Files
        </a>
        
        
    </li>
	
    <li class="">
        <a href="https://sourceforge.net/projects/mageck/reviews" class="ui-icon-tool-reviews-32">
            Reviews
        </a>
        
        
    </li>
	
    <li class="">
        <a href="https://sourceforge.net/projects/mageck/support" class="ui-icon-tool-support-32">
            Support
        </a>
        
        
    </li>
	
    <li class="selected">
        <a href="https://sourceforge.net/p/mageck/wiki/" class="ui-icon-tool-wiki-32">
            Wiki
        </a>
        
        
    </li>
	
    <li class="">
        <a href="https://sourceforge.net/p/mageck/discussion/" class="ui-icon-tool-discussion-32">
            Discussion
        </a>
        
        
    </li>
	
    <li class="">
        <a href="https://sourceforge.net/p/mageck/tickets/" class="ui-icon-tool-tickets-32">
            Tickets
        </a>
        
        
    </li>
	
    <li class="">
        <a href="https://sourceforge.net/p/mageck/admin/" class="ui-icon-tool-admin-32">
            Admin
        </a>
        
        
    </li>
	
</ul>

        
      </div>
      <div id="content_base">
      
			  
			    
          


<div id="sidebar">
  
    <form id="search" method="GET" action="https://sourceforge.net/p/mageck/wiki/search/">
      <input name="q" type="text" title="Search Wiki" placeholder="Search Wiki" tabindex="2">
    </form>
  
    
    
      
        
    
      <ul class="sidebarmenu">
      
    
    <li>
      <a href="https://sourceforge.net/p/mageck/wiki/" class="add_wiki_page"><b data-icon="+" class="ico ico-plus"></b> <span>Create Page</span></a>
    </li>
  
      
        
    
      </ul>
      
    
    
  
      
        
    
      <ul class="sidebarmenu">
      
    
    <li>
      <a href="https://sourceforge.net/p/mageck/wiki/" class="wiki_home"><span>Wiki Home</span></a>
    </li>
  
      
        
    
    <li>
      <a href="https://sourceforge.net/p/mageck/wiki/browse_pages/"><span>Browse Pages</span></a>
    </li>
  
      
        
    
    <li>
      <a href="https://sourceforge.net/p/mageck/wiki/browse_tags/"><span>Browse Labels</span></a>
    </li>
  
      
        
    
      </ul>
      
    
    
  
      
        
    
      <ul class="sidebarmenu">
      
    
    <li>
      <a href="https://sourceforge.net/p/mageck/wiki/subscribe?subscribe=True"><b data-icon="M" class="ico ico-mail"></b> <span>Subscribe to wiki</span></a>
    </li>
  
      
        
    
      </ul>
      
    
    
  
      
        
    
      <ul class="sidebarmenu">
      
    
    <li>
      <a href="https://sourceforge.net/p/mageck/wiki/markdown_syntax/"><span>Formatting Help</span></a>
    </li>
  
      
    
    
      </ul>
      
    
    
</div>
          
          
			  
			  
          
        
        <div class="grid-20 pad">
          <h2 class="dark title">Home
            <!-- actions -->
            <small>
            
  
    
      <a href="https://sourceforge.net/p/mageck/wiki/Home/edit" title="Edit"><b data-icon="p" class="ico ico-pencil" title="Edit"></b></a>
    
    <a href="https://sourceforge.net/p/mageck/wiki/Home/history" title="History"><b data-icon="N" class="ico ico-history" title="History"></b></a>
  
  
    
  
    <a href="https://sourceforge.net/p/mageck/wiki/Home/subscribe?subscribe=True" class="artifact_subscribe" title="Subscribe to this page"><b data-icon="M" class="ico ico-mail" title="Subscribe to this page"></b></a>
  

  
	<a href="https://sourceforge.net/p/mageck/wiki/Home/feed" title="RSS"><b data-icon="f" class="ico ico-feed" title="Feed"></b></a>
  <a href="https://sourceforge.net/p/mageck/wiki/search" title="Search"><b data-icon="s" class="ico ico-search" title="Search"></b></a>

            </small>
            <!-- /actions -->
          </h2>
		
  
<div class="editbox">
  
  <div class="grid-9 gravatar sm">
    <label class="simple">Authors: </label>
    
      
        
            <a href="https://sourceforge.net/u/davidliwei/">
                
                    
    
  
    <img src="./MAGeCK_Wiki_Home_files/b9cc1f9048ed9e437d36915feecd38c5" alt="Wei Li" title="Wei Li" class="emboss x16">
  
  
                
            </a>
        
    
    
  </div>
   
</div>


          <div>
            
            
  
<div class="markdown_content"><p><strong>Model-based Analysis of Genome-wide CRISPR-Cas9 Knockout (MAGeCK)</strong> is a computational tool to identify important genes from the recent genome-scale CRISPR-Cas9 knockout screens (or GeCKO) technology. </p>
<p>MAGeCK is developed and maintained by Wei Li and Han Xu from <a class="" href="http://liulab.dfci.harvard.edu/" rel="nofollow">Dr. Xiaole Shirley Liu's lab</a> at Department of Biostatistics and Computational Biology, Dana-Farber Cancer Institute and Harvard School of Public Health.</p>
<p>The MAGeCK algorithm is described in the following paper:</p>
<p><a class="" href="http://genomebiology.com/2014/15/12/554/abstract" rel="nofollow">Li, Wei et al. MAGeCK enables robust identification of essential genes from genome-scale CRISPR/Cas9 knockout screens. Genome Biology 15:554 (2014)</a></p>
<p>MAGeCK is a free, open source software under the BSD license.</p>
<p>This documentation includes the following items:</p>
<ul>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/install">Installation</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/demo">Tutorial</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/usage">Usage</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/input">File formats (input)</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/output">File formats (output)</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/libraries">Commonly used libraries</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/QA">Frequently Asked Questions</a></li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/history">Version history</a></li>
</ul>
<p>Take a look at the following videos to install and use MAGeCK:</p>
<ul>
<li>Installation</li>
</ul>
<div class="grid-20"><iframe height="270" src="./MAGeCK_Wiki_Home_files/saved_resource.html" width="480"></iframe>
<ul>
<li>Comparison between samples</li>
</ul>
<div class="grid-20"><iframe height="270" src="./MAGeCK_Wiki_Home_files/saved_resource.html" width="480"></iframe>
<div>
<div class="markdown_content"><h1 id="installation">Installation</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#installation">Installation</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#download-and-get-the-unzip-password">Download and get the unzip password</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#system-requirement">System requirement</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#installation_1">Installation</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#wzxhzdk8-manual-installation-wzxhzdk9"><s> Manual installation </s></a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#setting-up-the-environment-variables">Setting up the environment variables</a></li>
</ul>
</li>
</ul>
</div>
<h2 id="download-and-get-the-unzip-password">Download and get the unzip password</h2>
<p>The latest version of MAGeCK (0.4.1) can be downloaded here:<br>
</p><p><span class="download-button-533ec35981b24b2aced2594a" style="margin-bottom: 1em; display: block;">

    <style type="text/css" scoped="">.sfdl { width:165px;height:39px;padding:0;position:relative;font:bold 12px/1.17 sans-serif !important; border:1px solid #aaa;display:inline;float:left;text-decoration:none;-moz-box-shadow:rgba(255, 255, 255, 0.6) 0 1px 0 0;-webkit-box-shadow:rgba(255, 255, 255, 0.6) 0 1px 0 0;-o-box-shadow:rgba(255, 255, 255, 0.6) 0 1px 0 0;box-shadow:rgba(255, 255, 255, 0.6) 0 1px 0 0;-moz-border-radius:4px;-webkit-border-radius:4px;-o-border-radius:4px;-ms-border-radius:4px;-khtml-border-radius:4px;border-radius:4px;background:no-repeat #e5e5e5 8px center url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABEAAAANCAYAAABPeYUaAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAANFJREFUeNqcUosNgjAUbE0HcARGwAnqZxBxBCbQDXQE2UOkTIAj4AZsgPfMa1JeGqhecimU63HvvepxHFUq6mebY2nAdbi/Ur/hKg0IRv2H3WFvnX/Rj9qRcxH7A+BIjDK2eCYewQy8g2/WDJSk4w8xWDICz2ziUchyvMGLXIVJK9acE0+0VM53PIitE6bTcKJJT0wg6ESSHiwhHpbMDYszjhrDKcVkE5mO5dhZyrwNx72J2i9iGvMmODB372U/aCpV2NSlG0sGVbiBw2VM+BFgALYfPdL6UCEWAAAAAElFTkSuQmCC);background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABEAAAANCAYAAABPeYUaAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAANFJREFUeNqcUosNgjAUbE0HcARGwAnqZxBxBCbQDXQE2UOkTIAj4AZsgPfMa1JeGqhecimU63HvvepxHFUq6mebY2nAdbi/Ur/hKg0IRv2H3WFvnX/Rj9qRcxH7A+BIjDK2eCYewQy8g2/WDJSk4w8xWDICz2ziUchyvMGLXIVJK9acE0+0VM53PIitE6bTcKJJT0wg6ESSHiwhHpbMDYszjhrDKcVkE5mO5dhZyrwNx72J2i9iGvMmODB372U/aCpV2NSlG0sGVbiBw2VM+BFgALYfPdL6UCEWAAAAAElFTkSuQmCC), -webkit-gradient(linear, 0% 0%, 0% 100%, color-stop(0%, #fff), color-stop(100%, #ccc));background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABEAAAANCAYAAABPeYUaAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAANFJREFUeNqcUosNgjAUbE0HcARGwAnqZxBxBCbQDXQE2UOkTIAj4AZsgPfMa1JeGqhecimU63HvvepxHFUq6mebY2nAdbi/Ur/hKg0IRv2H3WFvnX/Rj9qRcxH7A+BIjDK2eCYewQy8g2/WDJSk4w8xWDICz2ziUchyvMGLXIVJK9acE0+0VM53PIitE6bTcKJJT0wg6ESSHiwhHpbMDYszjhrDKcVkE5mO5dhZyrwNx72J2i9iGvMmODB372U/aCpV2NSlG0sGVbiBw2VM+BFgALYfPdL6UCEWAAAAAElFTkSuQmCC), -moz-linear-gradient(top, #fff 0%, #ccc 100%);background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABEAAAANCAYAAABPeYUaAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAANFJREFUeNqcUosNgjAUbE0HcARGwAnqZxBxBCbQDXQE2UOkTIAj4AZsgPfMa1JeGqhecimU63HvvepxHFUq6mebY2nAdbi/Ur/hKg0IRv2H3WFvnX/Rj9qRcxH7A+BIjDK2eCYewQy8g2/WDJSk4w8xWDICz2ziUchyvMGLXIVJK9acE0+0VM53PIitE6bTcKJJT0wg6ESSHiwhHpbMDYszjhrDKcVkE5mO5dhZyrwNx72J2i9iGvMmODB372U/aCpV2NSlG0sGVbiBw2VM+BFgALYfPdL6UCEWAAAAAElFTkSuQmCC), linear-gradient(top, #fff 0%, #ccc 100%);background-position:8px center, 0 0;margin:0 10px 0px 0;overflow:hidden;color:#555 !important;}.sfdl:hover { cursor:pointer;text-decoration:none;background:8px center no-repeat #eee url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABEAAAANCAYAAABPeYUaAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAANFJREFUeNqcUosNgjAUbE0HcARGwAnqZxBxBCbQDXQE2UOkTIAj4AZsgPfMa1JeGqhecimU63HvvepxHFUq6mebY2nAdbi/Ur/hKg0IRv2H3WFvnX/Rj9qRcxH7A+BIjDK2eCYewQy8g2/WDJSk4w8xWDICz2ziUchyvMGLXIVJK9acE0+0VM53PIitE6bTcKJJT0wg6ESSHiwhHpbMDYszjhrDKcVkE5mO5dhZyrwNx72J2i9iGvMmODB372U/aCpV2NSlG0sGVbiBw2VM+BFgALYfPdL6UCEWAAAAAElFTkSuQmCC);}.sfdl-lite { background:0 center no-repeat #337733 none;color:white !important;height:auto;padding:0.5em 1em;text-align:center;-moz-border-radius:4px;-webkit-border-radius:4px;-o-border-radius:4px;-ms-border-radius:4px;-khtml-border-radius:4px;border-radius:4px;background-image:-webkit-gradient(linear, 0% 0%, 0% 100%, color-stop(0%, #339933), color-stop(100%, #337733));background-image:-moz-linear-gradient(top, #339933 0%, #337733 100%);background-image:linear-gradient(top, #339933 0%, #337733 100%);}.sfdl-lite:hover { background:0 center no-repeat #339933;}.sfdl span { border-radius:3px;-moz-border-radius:3px;-webkit-border-radius:3px;display:block;margin:3px;margin-left:35px;padding:4px;text-align:left;color:#FFF;background:#363;background-image:-moz-linear-gradient(center top, #393 0%, #373 100%);background-image:-webkit-gradient(linear, left top, left bottom, color-stop(0, #393),color-stop(1, #373));background-image:linear-gradient(top, #393 0%, #373 100%);overflow:hidden;height:25px;}.sfdl b { display:block;line-height:12.5px;}.sfdl small { display:block;line-height:12.5px;color:#ADC2AD; font-weight:normal;white-space:nowrap;overflow:hidden;font-size:77%;}.info-circle{ right:10px;bottom:16px;}.direct-dl{ font-size:10px;text-align:right;display:inline-block;width:100%;}</style>


<a href="https://sourceforge.net/projects/mageck/files/latest/download" title="Download /libraries/human_sam_library.csv.zip from SourceForge  - 858.8 kB" class="sfdl">
    <span>
        <b>Download</b>
        <small>human_sam_libraâ€¦zip</small>
        
    </span>
</a>
</span></p><p></p>
<hr>
<p>The zip file is encrypted, but you can get the password easily by one of the following options:</p>
<ul>
<li>Join our <a class="" href="https://groups.google.com/d/forum/mageck" rel="nofollow">MAGeCK Google group</a> and the password is shown on the top of the forum; or</li>
<li>Send an email to <a class="" href="mailto:mageck.help@gmail.com">mageck.help@gmail.com</a> with the subject "password", and you will get an automatic reply with password within 1 minute.</li>
</ul>
<p>You need to go to the Terminal to unzip and install the software. See the instructions for Installation below.</p>
<hr>
<h2 id="system-requirement">System requirement</h2>
<p>MAGeCK can be run on either Mac or Linux system. Since MAGeCK is written in Python and C, Python 2.7 (&gt;2.7) and a C compiler is needed. </p>
<p>MAGeCK also suggests users installing <a class="" href="http://www.numpy.org/" rel="nofollow">numpy</a> to calculate the negative binomial p value. However, in cases where <em>numpy</em> is not found, MAGeCK will use the normal p value instead. There may be slight differences between both methods.</p>
<h2 id="installation_1">Installation</h2>
<p>Since version 0.3, MAGeCK uses standard Python installation procedures (distutils) for compiling and installation of the software. </p>
<p>The installation procedure is extremely easy. First, download the source code, unzip it (assuming the current version is 0.4.1), type the unzip password, and go into the directory:</p>
<div class="codehilite"><pre><span class="n">unzip</span> <span class="n">mageck</span><span class="o">-</span><span class="mf">0.4.1</span><span class="p">.</span><span class="n">zip</span>
<span class="n">cd</span> <span class="n">mageck</span><span class="o">-</span><span class="mf">0.4</span>
</pre></div>
<p>After that, invoke python setup.py:</p>
<div class="codehilite"><pre><span class="n">python</span> <span class="n">setup</span><span class="p">.</span><span class="n">py</span> <span class="n">install</span>
</pre></div>
<p>And it is done! <b> If you want MAGeCK to be installed on your own directory </b>, use the following command instead:</p>
<div class="codehilite"><pre><span class="n">python</span> <span class="n">setup</span><span class="p">.</span><span class="n">py</span> <span class="n">install</span> <span class="o">--</span><span class="n">prefix</span><span class="o">=</span><span class="err">$</span><span class="n">HOME</span>
</pre></div>
<p>where <strong>$HOME</strong> is the root directory you want to install (usually the user home).</p>
<h2 id="wzxhzdk8-manual-installation-wzxhzdk9"><s> Manual installation </s></h2>
<p><b> The manual installation is deprecated since version 0.3. Please refer to the installation instructions above. </b></p>
<p>After downloading the source code, follow the instructions below for manual installation.</p>
<h2 id="setting-up-the-environment-variables">Setting up the environment variables</h2>
<p>Usually if you use the default Python installation options (<em>i.e.</em>, install MAGeCK into the system directory), you are all set. For those who installed MAGeCK into custom directories, there may be additional steps, including changing the environment variable. First you need to add the path of the mageck program to your <strong>PATH</strong> variable. For example, if you used the <strong>--prefix=$HOME</strong> option during installation, then set up the <strong>PATH</strong> variable by typing:</p>
<div class="codehilite"><pre><span class="n">export</span> <span class="n">PATH</span><span class="o">=</span><span class="err">$</span><span class="n">PATH</span><span class="o">:</span><span class="err">$</span><span class="n">HOME</span><span class="o">/</span><span class="n">bin</span>
</pre></div>
<p>You also need to add the path of the MAGeCK module to the <strong>PYTHONPATH</strong> variable. Again, his variable should be set as </p>
<div class="codehilite"><pre><span class="n">export</span> <span class="n">PYTHONPATH</span><span class="o">=</span><span class="err">$</span><span class="n">HOME</span><span class="o">/</span><span class="n">lib</span><span class="o">/</span><span class="n">python2</span><span class="mf">.7</span><span class="o">/</span><span class="n">site</span><span class="o">-</span><span class="n">packages</span><span class="o">:</span><span class="err">$</span><span class="n">PYTHONPATH</span>
</pre></div>
<p>To save the path configuration (so you don't have to type it every time), place the above command in your <em>~/.bashrc</em> (for Linux) or <em>~/.bash_profile</em> (for Mac).</p>
<p>Return to <a class="alink" href="https://sourceforge.net/p/mageck/wiki/Home">[Home]</a></p>
<hr></div>
</div>
<div>
<div class="markdown_content"><div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#tutorial">Tutorial</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#the-first-demo-starting-from-read-count-tables">The first demo: starting from read count tables</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#the-second-demo-starting-from-raw-fastq-files">The second demo: starting from raw fastq files</a></li>
</ul>
</li>
</ul>
</div>
<h1 id="tutorial">Tutorial</h1>
<p>Running MAGeCK is extremely easy and convenient. The <em>demo</em> folder contains two mini examples to go through all steps in MAGeCK. Simply execute the <em>sh</em> script in the command line in each example to run the demos.</p>
<p>Also check out the following videos in YouTube to learn how to install and run MAGeCK:</p>
<p><a class="" href="https://www.youtube.com/watch?v=0FLWSp1tdTI" rel="nofollow">Tutorial 1: Installation</a></p>
<p><a class="" href="https://www.youtube.com/watch?v=iIyP2w44ric" rel="nofollow">Tutorial 2: Comparison between samples</a></p>
<h2 id="the-first-demo-starting-from-read-count-tables">The first demo: starting from read count tables</h2>
<p>There is only one command line in the demo:</p>
<div class="codehilite"><pre>mageck <span class="nb">test</span> -k sample.txt -t HL60.final,KBM7.final -c HL60.initial,KBM7.initial  -n demo
</pre></div>
<p>The parameters are explained as follows.</p>
<table>
<thead>
<tr>
<th>Parameters</th>
<th>Meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td>mageck</td>
<td>The main portal of the MAGeCK program</td>
</tr>
<tr>
<td>test</td>
<td>A sub-command to ask MAGeCK to perform sgRNA and gene ranking based on provided read count tables</td>
</tr>
<tr>
<td>-k sgrna_count.txt</td>
<td>The provided read count table file. The format of the file is specified <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sgrna-read-count-file">here</a>.</td>
</tr>
<tr>
<td>-t HL60.final,KBM7.final</td>
<td>The treatment samples are defined as HL60.final,KBM7.final (or the 2nd and 3rd sample, starting from 0) in sgrna_count.txt. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sample-index">input files</a> for a detailed explanation.</td>
</tr>
<tr>
<td>-c HL60.initial,KBM7.initial</td>
<td>The control samples are defined as HL60.initial,KBM7.initial (or the 0th and 1st sample, starting from 0) in sgrna_count.txt. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sample-index">input files</a> for a detailed explanation.</td>
</tr>
<tr>
<td>-n demo</td>
<td>The prefix of the output files is demo, so you will expect the output files are: demo.sgrna_summary.txt, demo.gene_summary.txt, etc.</td>
</tr>
</tbody>
</table>
<p>An explanation of the output files can be found in the <a class="alink" href="https://sourceforge.net/p/mageck/wiki/output">[output]</a> page. For all available parameters, see the <a class="alink" href="https://sourceforge.net/p/mageck/wiki/usage">[usage]</a> page.</p>
<p>You can also specify the treatment and control samples using sample index. For example,</p>
<div class="codehilite"><pre>mageck <span class="nb">test</span> -k sgrna_count.txt -t 2,3 -c 0,1 -n demo
</pre></div>
<h2 id="the-second-demo-starting-from-raw-fastq-files">The second demo: starting from raw fastq files</h2>
<p>This demo shows an mini example of how to go through the whole pipeline from raw fastq files. In this example, we have fastq files from two conditions, and we would like to compare which gene and sgRNA is significant between conditions. The command line used in the <em>runmageck.sh</em> script is:</p>
<div class="codehilite"><pre>mageck run  -l library.txt -n demo --sample-label L1,CTRL -t CTRL -c L1 --fastq test1.fastq test2.fastq
</pre></div>
<p>The parameters are explained as follows.</p>
<table>
<thead>
<tr>
<th>Parameters</th>
<th>Meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td>mageck</td>
<td>The main portal of the MAGeCK program</td>
</tr>
<tr>
<td>run</td>
<td>A sub-command to ask MAGeCK to go through sgRNA counting, sgRNA and gene ranking from fastq files.</td>
</tr>
<tr>
<td>-l library.txt</td>
<td>The provided sgRNA information, including the sgRNA id, the sequence, and the gene it is targeting. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sgrna-library-file">input files</a> for a detailed explanation.</td>
</tr>
<tr>
<td>-n demo</td>
<td>The prefix of the output files is demo, so you will expect the output files are: demo.summary.txt, demo.gene.high.txt, etc.</td>
</tr>
<tr>
<td>--sample-label L1,CTRL</td>
<td>The labels of the two samples are L1 (test1.fastq) and CTRL (test2.fastq).</td>
</tr>
<tr>
<td>-t CTRL</td>
<td>The treatment samples are defined as CTRL. In another way, it is the sample in test2.fastq. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sample-index">input files</a> for a detailed explanation.</td>
</tr>
<tr>
<td>-c L1</td>
<td>The control samples are defined as L1. It is the sample in test1.fastq. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sample-index">input files</a> for a detailed explanation.</td>
</tr>
<tr>
<td>--fastq test1.fastq test2.fastq</td>
<td>The provided fastq file, separated by space. (Technical replicates of the same sample can also indicated using comma as a separator; for example, "sample1_replicate1.fastq,sample1_replicate2.fastq")</td>
</tr>
</tbody>
</table>
<p>You can also use</p>
<div class="codehilite"><pre>mageck run -l library.txt -n demo --sample-label L1,CTRL -t 1 -c 0 --fastq test1.fastq test2.fastq
</pre></div>
<p>Return to <a class="alink" href="https://sourceforge.net/p/mageck/wiki/Home">[Home]</a></p>
<hr></div>
</div>
<div>
<div class="markdown_content"><h1 id="usage">Usage</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#usage">Usage</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#run">run</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#test">test</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#count">count</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#pathway">pathway</a></li>
</ul>
</li>
</ul>
</div>
<p>The main portal of MAGeCK is the <strong>mageck.py</strong> script, which includes 3 different subprograms:</p>
<ul>
<li>run: collect sgRNA read counts from read mapping files (sam format), and perform sgRNA and gene ranking.</li>
<li>count: only collect sgRNA read counts from read mapping files (sam format).</li>
<li>test: given a table of read counts, perform the sgRNA and gene ranking.</li>
<li>pathway: given a ranked gene list, test whether one pathway is enriched.</li>
</ul>
<h2 id="run">run</h2>
<p>The parameters for the run sub-command are included in <em>test</em> and <em>count</em> sub-command. See both sub-commands for more details.</p>
<h2 id="test">test</h2>
<p>This subcommand tests and ranks sgRNAs and genes based on the read count tables provided.</p>
<p><em>usage</em>:</p>
<div class="codehilite"><pre><span class="n">usage</span><span class="o">:</span> <span class="n">mageck</span> <span class="n">test</span> <span class="o">[-</span><span class="n">h</span><span class="o">]</span> <span class="o">-</span><span class="n">k</span> <span class="n">COUNT_TABLE</span> <span class="o">-</span><span class="n">t</span> <span class="n">TREATMENT_ID</span> <span class="o">[-</span><span class="n">c</span> <span class="n">CONTROL_ID</span><span class="o">]</span>
               <span class="o">[-</span><span class="n">n</span> <span class="n">OUTPUT_PREFIX</span><span class="o">]</span> <span class="o">[--</span><span class="n">norm</span><span class="o">-</span><span class="n">method</span> <span class="o">{</span><span class="n">none</span><span class="o">,</span><span class="n">median</span><span class="o">,</span><span class="n">total</span><span class="o">}]</span>
               <span class="o">[--</span><span class="n">normcounts</span><span class="o">-</span><span class="n">to</span><span class="o">-</span><span class="n">file</span><span class="o">]</span>
               <span class="o">[--</span><span class="n">gene</span><span class="o">-</span><span class="n">test</span><span class="o">-</span><span class="n">fdr</span><span class="o">-</span><span class="n">threshold</span> <span class="n">GENE_TEST_FDR_THRESHOLD</span><span class="o">]</span>
               <span class="o">[--</span><span class="n">adjust</span><span class="o">-</span><span class="n">method</span> <span class="o">{</span><span class="n">fdr</span><span class="o">,</span><span class="n">holm</span><span class="o">}]</span> <span class="o">[--</span><span class="n">variance</span><span class="o">-</span><span class="n">from</span><span class="o">-</span><span class="n">all</span><span class="o">-</span><span class="n">samples</span><span class="o">]</span>
               <span class="o">[--</span><span class="n">sort</span><span class="o">-</span><span class="n">criteria</span> <span class="o">{</span><span class="n">neg</span><span class="o">,</span><span class="n">pos</span><span class="o">}]</span> <span class="o">[--</span><span class="n">keep</span><span class="o">-</span><span class="n">tmp</span><span class="o">]</span>
</pre></div>
<p><em>required arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-k COUNT_TABLE, --count-table COUNT_TABLE</td>
<td>Provide a tab-separated count table instead of sam files. Each line in the table should include sgRNA name (1st column), targeting gene (2nd column) and read counts in each sample. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sgrna-read-count-file">input/#sgrna-read-count-file</a> for a detailed description.</td>
</tr>
<tr>
<td>-t TREATMENT_ID, --treatment-id TREATMENT_ID</td>
<td>Sample label or sample index (0 as the first sample) in the count table as treatment experiments, separated by comma (,). If sample label is provided, the labels must match the labels in the first line of the count table; for example, "HL60.final,KBM7.final". For sample index, "0,2" means the 1st and 3rd samples are treatment experiments. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sample-index">input/#sample-index</a> for a detailed description.</td>
</tr>
</tbody>
</table>
<p><em>optional arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-h, --help</td>
<td>show this help message and exit</td>
</tr>
<tr>
<td>-c CONTROL_ID, --control-id CONTROL_ID</td>
<td>Sample label or sample index in the count table as control experiments, separated by comma (,). Default is all the samples not specified in treatment experiments. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sample-index">input/#sample-index</a> for a detailed description.</td>
</tr>
<tr>
<td>-n OUTPUT_PREFIX, --output-prefix OUTPUT_PREFIX</td>
<td>The prefix of the output file(s). Default sample1.</td>
</tr>
<tr>
<td>--norm-method {none,median,total}</td>
<td>Method for normalization, default median.</td>
</tr>
<tr>
<td>--normcounts-to-file</td>
<td>Write normalized read counts to file ({output-prefix}.normalized.txt).</td>
</tr>
<tr>
<td>--gene-test-fdr-threshold GENE_TEST_FDR_THRESHOLD</td>
<td>FDR threshold for gene test, default 0.05.</td>
</tr>
<tr>
<td>--adjust-method {fdr,holm}</td>
<td>Method for p-value adjustment, including false discovery rate (fdr) or holm's method (holm). Default fdr.</td>
</tr>
<tr>
<td>--variance-from-all-samples</td>
<td>Estimate the variance from all samples, instead of from only control samples. Use this option only if you believe there are relatively few essential sgRNAs or genes between control and treatment samples.</td>
</tr>
<tr>
<td>--sort-criteria {neg,pos}</td>
<td>Sorting criteria, either by negative selection (neg) or positive selection (pos). Default negative selection.</td>
</tr>
<tr>
<td>--keep-tmp</td>
<td>Keep intermediate files.</td>
</tr>
</tbody>
</table>
<h2 id="count">count</h2>
<p>This subcommand collects sgRNA read count information from fastq files. The output count tables can be used directly in the <em>test</em> subcommand.</p>
<p><em>usage</em>: </p>
<div class="codehilite"><pre><span class="n">usage</span><span class="o">:</span> <span class="n">mageck</span> <span class="n">count</span> <span class="o">[-</span><span class="n">h</span><span class="o">]</span> <span class="o">[-</span><span class="n">l</span> <span class="n">LIST_SEQ</span><span class="o">]</span> <span class="o">[-</span><span class="n">n</span> <span class="n">OUTPUT_PREFIX</span><span class="o">]</span>
                <span class="o">[--</span><span class="n">sample</span><span class="o">-</span><span class="n">label</span> <span class="n">SAMPLE_LABEL</span><span class="o">]</span> <span class="o">[--</span><span class="n">trim</span><span class="o">-</span><span class="mi">5</span> <span class="n">TRIM_5</span><span class="o">]</span>
                <span class="o">[--</span><span class="n">sgrna</span><span class="o">-</span><span class="n">len</span> <span class="n">SGRNA_LEN</span><span class="o">]</span> <span class="o">[--</span><span class="n">count</span><span class="o">-</span><span class="n">n</span><span class="o">]</span> <span class="o">--</span><span class="n">fastq</span> <span class="n">FASTQ</span> <span class="o">[</span><span class="n">FASTQ</span> <span class="o">...]</span>
</pre></div>
<p><em>required arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>--fastq</td>
<td>Sample fastq files, separated by space; use comma (,)  to indicate technical replicates of the same sample. For example, "--fastq sample1_replicate1.fastq,sample1_replicate2.fastq sample2_replicate1.fastq,sample2_replicate2.fastq" indicates two samples with 2 technical replicates for each sample.</td>
</tr>
</tbody>
</table>
<p><em>optional arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-h, --help</td>
<td>show this help message and exit</td>
</tr>
<tr>
<td>-l LIST_SEQ, --list-seq LIST_SEQ</td>
<td>A file containing list of sgRNA names, the sequences and target genes, either in .txt or in .csv format. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sgrna-library-file">input/#sgrna-library-file</a> for more details. If this file is not provided, mageck will count all possible sgRNAs in the fastq.</td>
</tr>
<tr>
<td>-n OUTPUT_PREFIX, --output-prefix OUTPUT_PREFIX</td>
<td>The prefix of the output file(s). Default sample1.</td>
</tr>
<tr>
<td>--sample-label SAMPLE_LABEL</td>
<td>Sample labels, separated by comma (,). Must be equal to the number of samples provided (in --fastq option). Default "sample1,sample2,...".</td>
</tr>
<tr>
<td>--trim-5 TRIM_5</td>
<td>Length of trimming the 5' of the reads. Default 0</td>
</tr>
<tr>
<td>--sgrna-len SGRNA_LEN</td>
<td>Length of the sgRNA. Default 20</td>
</tr>
<tr>
<td>--count-n</td>
<td>Count sgRNAs with Ns. By default, sgRNAs containing Ns will be discarded.</td>
</tr>
</tbody>
</table>
<h2 id="pathway">pathway</h2>
<p>MAGeCK can also invoke RRA to test if a pathway is enriched in one particular gene ranking. </p>
<p><em>usage</em>:</p>
<div class="codehilite"><pre><span class="n">usage</span><span class="o">:</span> <span class="n">mageck</span> <span class="n">pathway</span> <span class="o">[-</span><span class="n">h</span><span class="o">]</span> <span class="o">--</span><span class="n">gene</span><span class="o">-</span><span class="n">ranking</span> <span class="n">GENE_RANKING</span> <span class="o">--</span><span class="n">gmt</span><span class="o">-</span><span class="n">file</span> <span class="n">GMT_FILE</span>
                  <span class="o">[-</span><span class="n">n</span> <span class="n">OUTPUT_PREFIX</span><span class="o">]</span>
</pre></div>
<p><em>required arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>--gene-ranking GENE_RANKING</td>
<td>The gene ranking file generated by the gene test step.</td>
</tr>
<tr>
<td>--gmt-file GMT_FILE</td>
<td>The pathway file in GMT format. See <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#pathway-file-gmt">input/#pathway-file-gmt</a> for more details of the GMT file format.</td>
</tr>
</tbody>
</table>
<p><em>optional arguments</em>:</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-h, --help</td>
<td>show this help message and exit</td>
</tr>
<tr>
<td>--single-ranking</td>
<td>The provided file is a (single) gene ranking file, either positive or negative selection. Only one enrichment comparison will be performed.</td>
</tr>
<tr>
<td>-n OUTPUT_PREFIX, --output-prefix OUTPUT_PREFIX</td>
<td>The prefix of the output file(s). Default sample1.</td>
</tr>
<tr>
<td>--sort-criteria {neg,pos}</td>
<td>Sorting criteria, either by negative selection (neg) or positive selection (pos). Default negative selection.</td>
</tr>
<tr>
<td>--keep-tmp</td>
<td>Keep intermediate files.</td>
</tr>
</tbody>
</table>
<p>Return to <a class="alink" href="https://sourceforge.net/p/mageck/wiki/Home">[Home]</a></p>
<hr></div>
</div>
<div>
<div class="markdown_content"><h1 id="input-file-specification">Input file specification</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#input-file-specification">Input file specification</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#sgrna-read-count-file">sgRNA read count file</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#sample-index">Sample index</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#sgrna-library-file">sgRNA library file</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#pathway-file-gmt">pathway file (gmt)</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#sgrnagene-mapping-file-depreciated-after-version-03">sgRNA/gene mapping file (depreciated after version 0.3)</a></li>
</ul>
</li>
</ul>
</div>
<h2 id="sgrna-read-count-file">sgRNA read count file</h2>
<p>The sgRNA read count file will be used in <em>-k</em> parameter in the <em>test</em> or <em>run</em> sub-command.</p>
<p>The read count file should list the names of the sgRNA, the gene it is targeting, followed by the read counts in each sample. Each item should be separated by the tab ('\t'). A header line is optional. For example:</p>
<div class="codehilite"><pre><span class="n">sgRNA</span>           <span class="n">gene</span>    <span class="n">HL60</span><span class="p">.</span><span class="n">initial</span>    <span class="n">KBM7</span><span class="p">.</span><span class="n">initial</span>    <span class="n">HL60</span><span class="p">.</span><span class="n">final</span>      <span class="n">KBM7</span><span class="p">.</span><span class="n">final</span>
<span class="n">A1CF_m52595977</span>  <span class="n">A1CF</span>    <span class="mi">213</span>     <span class="mi">274</span>     <span class="mi">883</span>     <span class="mi">175</span>
<span class="n">A1CF_m52596017</span>  <span class="n">A1CF</span>    <span class="mi">294</span>     <span class="mi">412</span>     <span class="mi">1554</span>    <span class="mi">1891</span>
<span class="n">A1CF_m52596056</span>  <span class="n">A1CF</span>    <span class="mi">421</span>     <span class="mi">368</span>     <span class="mi">566</span>     <span class="mi">759</span>
<span class="n">A1CF_m52603842</span>  <span class="n">A1CF</span>    <span class="mi">274</span>     <span class="mi">243</span>     <span class="mi">314</span>     <span class="mi">855</span>
<span class="n">A1CF_m52603847</span>  <span class="n">A1CF</span>    <span class="mi">0</span>       <span class="mi">50</span>      <span class="mi">145</span>     <span class="mi">266</span>
</pre></div>
<p>The <em>count</em> sub-command will output the read count file like this.</p>
<h2 id="sample-index">Sample index</h2>
<p>In the <em>-t/--treatment-id</em>, <em>-c/--control-id</em> parameters, you can use either sample label or sample index to specify samples. If sample label is used, the labels <a class="alink notfound" href="https://sourceforge.net/p/mageck/wiki/must">[must]</a> match the sample labels in the first line of the count table. For example, "HL60.final,KBM7.final".</p>
<p>You can also use sample index to specify samples. The index of the sample is the order it appears in the sgRNA read count file, starting from 0. The index is used in the <em>-t/--treatment-id</em>, <em>-c/--control-id</em> parameters. In the example above, there are four samples, and the index of each sample is as follows:</p>
<table>
<thead>
<tr>
<th>sample</th>
<th>index</th>
</tr>
</thead>
<tbody>
<tr>
<td>HL60.initial</td>
<td>0</td>
</tr>
<tr>
<td>KBM7.initial</td>
<td>1</td>
</tr>
<tr>
<td>HL60.final</td>
<td>2</td>
</tr>
<tr>
<td>KBM7.final</td>
<td>3</td>
</tr>
</tbody>
</table>
<h2 id="sgrna-library-file">sgRNA library file</h2>
<p>When starting from fastq files, MAGeCK needs to know the sgRNA sequence and its targeting gene. Such information is provided in the sgRNA library file, and can be specified by the <em>-l/--list-seq</em> option in <strong>run</strong> or <strong>count</strong> subcommand.</p>
<p><strong>The sgRNA library file can be provided either in .txt format or in .csv format.</strong> There are three columns in the library file: the sgRNA ID, the sequence, and the gene it is targeting. One example of the library file is provided as <em>library.txt</em> in demo2:</p>
<div class="codehilite"><pre><span class="n">s_10007</span> <span class="n">TGTTCACAGTATAGTTTGCC</span>    <span class="n">CCNA1</span>
<span class="n">s_10008</span> <span class="n">TTCTCCCTAATTGCTTGCTG</span>    <span class="n">CCNA1</span>
<span class="n">s_10027</span> <span class="n">ACATGTTGCTTCCCCTTGCA</span>    <span class="n">CCNC</span>
</pre></div>
<p>If provided in .csv format, the file will look like:</p>
<div class="codehilite"><pre><span class="n">s_10007</span><span class="p">,</span><span class="n">TGTTCACAGTATAGTTTGCC</span><span class="p">,</span><span class="n">CCNA1</span>
<span class="n">s_10008</span><span class="p">,</span><span class="n">TTCTCCCTAATTGCTTGCTG</span><span class="p">,</span><span class="n">CCNA1</span>
<span class="n">s_10027</span><span class="p">,</span><span class="n">ACATGTTGCTTCCCCTTGCA</span><span class="p">,</span><span class="n">CCNC</span>
</pre></div>
<h2 id="pathway-file-gmt">pathway file (gmt)</h2>
<p>The GMT file format stores the pathway information and is consistent with the GMT file in Gene Set Enrichment Analysis (GSEA). The details of the GMT format can be found at <a class="" href="http://www.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29" rel="nofollow">GSEA website</a>.</p>
<p>You can also download different pathway files directly from GSEA <a class="" href="http://www.broadinstitute.org/gsea/downloads.jsp" rel="nofollow">MSigDB</a> database. They can be used directly by MAGeCK. </p>
<h2 id="sgrnagene-mapping-file-depreciated-after-version-03">sgRNA/gene mapping file (depreciated after version 0.3)</h2>
<p>The sgRNA/gene mapping file will be used in the <em>--gene-test</em> parameter in the <em>test</em> or <em>run</em> sub-command.</p>
<p>This file should list the names of the sgRNAs and their corresponding genes, separated by the tab ('\t'). For example:</p>
<div class="codehilite"><pre><span class="n">A1CF_m52595977</span>  <span class="n">A1CF</span>
<span class="n">A1CF_m52596017</span>  <span class="n">A1CF</span>
<span class="n">A1CF_m52596056</span>  <span class="n">A1CF</span>
<span class="n">A1CF_m52603842</span>  <span class="n">A1CF</span>
<span class="n">A1CF_m52603847</span>  <span class="n">A1CF</span>
<span class="n">A1CF_p52595870</span>  <span class="n">A1CF</span>
<span class="n">A1CF_p52595881</span>  <span class="n">A1CF</span>
<span class="n">A1CF_p52596023</span>  <span class="n">A1CF</span>
</pre></div>
<p>Return to <a class="alink" href="https://sourceforge.net/p/mageck/wiki/Home">[Home]</a></p>
<hr></div>
</div>
<div>
<div class="markdown_content"><h1 id="output-file-specification">Output file specification</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#output-file-specification">Output file specification</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#sgrna_summary_txt">sgrna_summary_txt</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#gene_summary_txt">gene_summary_txt</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#pathway_summary_txt">pathway_summary_txt</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#log">log</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#intermediate-file-formats">Intermediate file formats</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#gene_txt">gene_txt</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<p>The output of the MAGeCK consists of the following files:</p>
<ul>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#sgrna_summary_txt">sgrna_summary.txt</a>: The sgRNA ranking results.</li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#gene_summary_txt">gene_summary.txt</a>: The gene ranking results.</li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#pathway_summary_txt">pathway_summary.txt</a>: The pathway ranking results.</li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#log">log</a>: The logging information during the running.</li>
</ul>
<p>Other file formats are intermediate files, including:</p>
<ul>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#gene_txt">.gene.high.txt</a>:  The gene ranking results (positively selected genes).</li>
<li><a class="" href="https://sourceforge.net/p/mageck/wiki/Home/#gene_txt">.gene.low.txt</a>:   The gene ranking results (negatively selected genes).</li>
</ul>
<h2 id="sgrna_summary_txt">sgrna_summary_txt</h2>
<p>An example of the sgRNA ranking results is as follows:</p>
<div class="codehilite"><pre><span class="n">sgrna</span>   <span class="n">Gene</span>   <span class="n">control_count</span>   <span class="n">treatment_count</span> <span class="n">control_mean</span>    <span class="n">treat_mean</span>      <span class="n">control_var</span>     <span class="n">adj_var</span> <span class="n">score</span>   <span class="n">p</span><span class="p">.</span><span class="n">low</span>   <span class="n">p</span><span class="p">.</span><span class="n">high</span>  <span class="n">p</span><span class="p">.</span><span class="n">twosided</span>      <span class="n">FDR</span>     <span class="n">high_in_treatment</span>
<span class="n">INO80B_m74682554</span>   <span class="n">INO80B</span>        <span class="mf">0.0</span><span class="o">/</span><span class="mf">0.0</span> <span class="mf">1220.1598778</span><span class="o">/</span><span class="mf">1476.14096301</span>      <span class="mf">0.810860655738</span>  <span class="mf">1348.15042041</span>   <span class="mf">0.0</span>     <span class="mf">19.0767988005</span>   <span class="mf">308.478081895</span>   <span class="mf">1.0</span>     <span class="mf">1.11022302463e-16</span>       <span class="mf">2.22044604925e-16</span>       <span class="mf">1.57651669497e-14</span>       <span class="n">True</span>
<span class="n">NHS_p17705966</span>   <span class="n">NHS</span>   <span class="mf">1.62172131148</span><span class="o">/</span><span class="mf">3.90887850467</span>     <span class="mf">2327.09368635</span><span class="o">/</span><span class="mf">1849.95115143</span>     <span class="mf">2.76529990807</span>   <span class="mf">2088.52241889</span>   <span class="mf">2.6155440132</span>    <span class="mf">68.2450168229</span>   <span class="mf">252.480744404</span>   <span class="mf">1.0</span>     <span class="mf">1.11022302463e-16</span>       <span class="mf">2.22044604925e-16</span>       <span class="mf">1.57651669497e-14</span>       <span class="n">True</span>
</pre></div>
<p>The contents of each column are as follows.</p>
<table>
<thead>
<tr>
<th>Column</th>
<th>Content</th>
</tr>
</thead>
<tbody>
<tr>
<td>sgrna</td>
<td>sgRNA ID</td>
</tr>
<tr>
<td>Gene</td>
<td>The targeting gene</td>
</tr>
<tr>
<td>control_count</td>
<td>Normalized read counts in control samples</td>
</tr>
<tr>
<td>treatment_count</td>
<td>Normalized read counts in treatment samples</td>
</tr>
<tr>
<td>control_mean</td>
<td>Mean read counts in control samples</td>
</tr>
<tr>
<td>treat_mean</td>
<td>Mean read counts in treatment samples</td>
</tr>
<tr>
<td>control_var</td>
<td>The raw variance in control samples</td>
</tr>
<tr>
<td>adj_var</td>
<td>The adjusted variance in control samples</td>
</tr>
<tr>
<td>score</td>
<td>The score of this sgRNA</td>
</tr>
<tr>
<td>p.low</td>
<td>p-value (lower tail)</td>
</tr>
<tr>
<td>p.high</td>
<td>p-value (higher tail)</td>
</tr>
<tr>
<td>p.twosided</td>
<td>p-value (two sided)</td>
</tr>
<tr>
<td>FDR</td>
<td>false discovery rate</td>
</tr>
<tr>
<td>high_in_treatment</td>
<td>Whether the abundance is higher in treatment samples</td>
</tr>
</tbody>
</table>
<h2 id="gene_summary_txt">gene_summary_txt</h2>
<p>An example of the gene ranking file (.gene.high.txt or .gene.low.txt) is as follows:</p>
<div class="codehilite"><pre><span class="n">id</span>      <span class="n">num</span><span class="p">.</span><span class="n">neg</span> <span class="n">p</span><span class="p">.</span><span class="n">neg</span>   <span class="n">fdr</span><span class="p">.</span><span class="n">neg</span> <span class="n">rank</span><span class="p">.</span><span class="n">neg</span>        <span class="n">num</span><span class="p">.</span><span class="n">pos</span> <span class="n">p</span><span class="p">.</span><span class="n">pos</span>   <span class="n">fdr</span><span class="p">.</span><span class="n">pos</span> <span class="n">rank</span><span class="p">.</span><span class="n">pos</span>
<span class="n">RPL18A</span>  <span class="mi">6</span>       <span class="mf">6.9363e-11</span>      <span class="mf">0.009901</span>        <span class="mi">2</span>       <span class="mi">6</span>       <span class="mf">1.0</span>     <span class="mf">1.0</span>     <span class="mi">21813</span>
<span class="n">CASP8AP2</span>        <span class="mi">6</span>       <span class="mf">6.9916e-07</span>      <span class="mf">0.029703</span>        <span class="mi">3</span>       <span class="mi">6</span>       <span class="mf">1.0</span>     <span class="mf">1.0</span>     <span class="mi">21812</span>
<span class="n">NBPF24</span>  <span class="mi">3</span>       <span class="mf">6.9959e-07</span>      <span class="mf">0.029703</span>        <span class="mi">4</span>       <span class="mi">3</span>       <span class="mf">1.0</span>     <span class="mf">1.0</span>     <span class="mi">21811</span>
</pre></div>
<p>The contents of each column is as follows.</p>
<table>
<thead>
<tr>
<th>Column</th>
<th>Content</th>
</tr>
</thead>
<tbody>
<tr>
<td>id</td>
<td>Gene ID</td>
</tr>
<tr>
<td>num.neg</td>
<td>The number of targeting sgRNAs for each gene in negative selection</td>
</tr>
<tr>
<td>p.neg</td>
<td>The raw p-value of this gene in negative selection</td>
</tr>
<tr>
<td>fdr.neg</td>
<td>The false discovery rate of this gene in negative selection</td>
</tr>
<tr>
<td>rank.neg</td>
<td>The ranking of this gene in negative selection</td>
</tr>
<tr>
<td>num.pos</td>
<td>The number of targeting sgRNAs for each gene in positive selection (usually the same as num.neg)</td>
</tr>
<tr>
<td>p.pos</td>
<td>The raw p-value of this gene in positive selection</td>
</tr>
<tr>
<td>fdr.pos</td>
<td>The false discovery rate of this gene in positive selection</td>
</tr>
<tr>
<td>rank.pos</td>
<td>The ranking of this gene in positive selection</td>
</tr>
</tbody>
</table>
<p>Genes are ranked by the p.neg field (by default). If you need a ranking by the p.pos, you can use the --sort-criteria option.</p>
<h2 id="pathway_summary_txt">pathway_summary_txt</h2>
<p>The output of the pathway summary is similar to the gene summary. Here is an example:</p>
<div class="codehilite"><pre><span class="n">id</span>      <span class="n">num</span><span class="p">.</span><span class="n">neg</span> <span class="n">p</span><span class="p">.</span><span class="n">neg</span>   <span class="n">fdr</span><span class="p">.</span><span class="n">neg</span> <span class="n">rank</span><span class="p">.</span><span class="n">neg</span>        <span class="n">num</span><span class="p">.</span><span class="n">pos</span> <span class="n">p</span><span class="p">.</span><span class="n">pos</span>   <span class="n">fdr</span><span class="p">.</span><span class="n">pos</span> <span class="n">rank</span><span class="p">.</span><span class="n">pos</span>
<span class="n">KEGG_RIBOSOME</span>   <span class="mi">87</span>      <span class="mf">4.4271e-25</span>      <span class="mf">0.0011</span>  <span class="mi">1</span>       <span class="mi">87</span>      <span class="mf">0.99995</span> <span class="mf">1.0</span>     <span class="mi">187</span>
<span class="n">KEGG_ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM</span> <span class="mi">32</span>      <span class="mf">0.025868</span>        <span class="mf">0.663079</span>        <span class="mi">2</span>      <span class="mi">32</span>      <span class="mf">0.99364</span> <span class="mf">1.0</span>     <span class="mi">186</span>
</pre></div>
<p>This table shows a pathway <em>KEGG_RIBOSOME</em> has 87 genes, its RRA p value is 4.42e-25 (negative selection), FDR 0.011, and its ranking is 1. This shows the genes in this pathway (i.e., ribosomal genes) are strongly negatively selected, which is expected in negative selection CRISPR experiments.</p>
<h2 id="log">log</h2>
<p>This file includes the logging information during the execution.</p>
<h2 id="intermediate-file-formats">Intermediate file formats</h2>
<p>These files will be automatically deleted after the completion of each command. To keep these files, use the "--keep-tmp" option during the execution.</p>
<h3 id="gene_txt">gene_txt</h3>
<p>An example of the gene ranking file (.gene.high.txt or .gene.low.txt) is as follows:</p>
<div class="codehilite"><pre> <span class="n">group_id</span>        <span class="err">#</span><span class="n">_items_in_group</span>        <span class="n">lo_value</span>        <span class="n">FDR</span>
 <span class="n">RPL3</span>    <span class="mi">93</span>      <span class="mf">4.9169e-36</span>      <span class="mf">0.000080</span>
 <span class="n">RPL8</span>    <span class="mi">67</span>      <span class="mf">1.8232e-24</span>      <span class="mf">0.000080</span>
 <span class="n">RPS2</span>    <span class="mi">61</span>      <span class="mf">1.6928e-20</span>      <span class="mf">0.000080</span>
 <span class="n">RPS18</span>   <span class="mi">40</span>      <span class="mf">1.0152e-18</span>      <span class="mf">0.000080</span>
</pre></div>
<p>The contents of each column is as follows.</p>
<table>
<thead>
<tr>
<th>Column</th>
<th>Content</th>
</tr>
</thead>
<tbody>
<tr>
<td>group_id</td>
<td>Gene ID</td>
</tr>
<tr>
<td>#_items_in_group</td>
<td>The number of targeting sgRNAs for each gene</td>
</tr>
<tr>
<td>lo_value</td>
<td>The raw p-value</td>
</tr>
<tr>
<td>FDR</td>
<td>The false discovery rate</td>
</tr>
</tbody>
</table>
<p>Return to <a class="alink" href="https://sourceforge.net/p/mageck/wiki/Home">[Home]</a></p>
<hr></div>
</div>
<div>
<div class="markdown_content"><h1 id="commonly-used-libraries">Commonly used libraries</h1>
<p><a class="" href="https://sourceforge.net/projects/mageck/files/libraries">Download frequently used libraries</a></p>
<p>For your convenience, we provide a set of library files that are ready to be used in MAGeCK (in the -l/--list-seq option of the count command) in the <a class="" href="https://sourceforge.net/projects/mageck/files/libraries">libraries</a> folder. You can also create your own library files, see <a class="" href="https://sourceforge.net/p/mageck/wiki/input/#sgrna-library-file">sgrna-library-file</a> for more details.</p>
<table>
<thead>
<tr>
<th>File</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>Human_GeCKOv2_Library_A_3_mageck.csv.zip</td>
<td>Human <a class="" href="http://www.addgene.org/crispr/libraries/geckov2" rel="nofollow">GeCKO v2</a> half-library A (can be used in either 1- or 2-plasmid systems)</td>
</tr>
<tr>
<td>Human_GeCKOv2_Library_B_1_mageck.csv.zip</td>
<td>Human GeCKO v2 half-library B</td>
</tr>
<tr>
<td>Human_GeCKOv2_Library_combine.csv.zip</td>
<td>Human GeCKO v2 combined library of A and B</td>
</tr>
<tr>
<td>mouse_geckov2_library_a_2_mageck.csv.zip</td>
<td>Mouse GeCKO v2 half-library A (can be used in either 1- or 2-plasmid systems)</td>
</tr>
<tr>
<td>mouse_geckov2_library_b_1_mageck.csv.zip</td>
<td>Mouse GeCKO v2 half-library B</td>
</tr>
<tr>
<td>mouse_geckov2_library_combine.csv.zip</td>
<td>Mouse GeCKO v2 combined library of A and B</td>
</tr>
<tr>
<td>GeCKOv1.txt.zip</td>
<td>GeCKO v1 library file (from the GeCKO <a class="" href="http://www.ncbi.nlm.nih.gov/pubmed/24336571" rel="nofollow">Science paper</a>)</td>
</tr>
<tr>
<td>human_sam_library.csv.zip</td>
<td>Human <a class="" href="http://www.addgene.org/crispr/libraries/sam" rel="nofollow">Synergistic Activation Mediator (SAM)</a> pooled library (CRISPRa library), generated by Feng Zhang laboratory.</td>
</tr>
</tbody>
</table></div>
</div>
<div>
<div class="markdown_content"><h1 id="q-and-a">Q and A</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#q-and-a">Q and A</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#generating-read-count-tables-from-fastq-file">Generating read count tables from fastq file</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#installation-problems">Installation problems</a></li>
</ul>
</li>
</ul>
</div>
<h2 id="generating-read-count-tables-from-fastq-file">Generating read count tables from fastq file</h2>
<p><strong>Q</strong>: How to deal with biological replicates and technical replicates?<br>
<strong>A</strong>: Usually you can pool the read counts for technical replicates of the same sample. To do this, use comma (,) to separate the fastq files of the technical replicates from the same sample in the --fastq option. For example, "--fastq sample1_replicate1.fastq,sample1_replicate2.fastq sample2_replicate1.fastq,sample2_replicate2.fastq" indicates two samples with 2 technical replicates for each sample.<br>
For biological replicates, treat them as separate samples and use them together when doing the comparison; so MAGeCK can analyze the variance of these samples. For example in the test command, "-t sample1_bio_replicate1,sample1_bio_replicate2 -c sample2_bio_replicate1,sample2_bio_replicate2" compares 2 samples (with 2 biological replicates in each sample).</p>
<p><strong>Q</strong>: How do I get the simple statistics of the fastq files, like, the total reads sequenced, the number of reads mapped to the sequences in the given library?<br>
<strong>A</strong>: It's in the log file (for run and count command). For example:</p>
<div class="codehilite"><pre><span class="n">INFO</span>  <span class="err">@</span> <span class="n">Mon</span><span class="p">,</span> <span class="mo">02</span> <span class="n">Feb</span> <span class="mi">2015</span> <span class="mi">08</span><span class="o">:</span><span class="mi">12</span><span class="o">:</span><span class="mi">15</span><span class="o">:</span> <span class="n">Summary</span> <span class="n">of</span> <span class="n">file</span> <span class="n">sample1_R1</span><span class="p">.</span><span class="n">fastq</span><span class="o">:</span> 
<span class="n">INFO</span>  <span class="err">@</span> <span class="n">Mon</span><span class="p">,</span> <span class="mo">02</span> <span class="n">Feb</span> <span class="mi">2015</span> <span class="mi">08</span><span class="o">:</span><span class="mi">12</span><span class="o">:</span><span class="mi">15</span><span class="o">:</span> <span class="n">reads</span>        <span class="mi">45631055</span> 
<span class="n">INFO</span>  <span class="err">@</span> <span class="n">Mon</span><span class="p">,</span> <span class="mo">02</span> <span class="n">Feb</span> <span class="mi">2015</span> <span class="mi">08</span><span class="o">:</span><span class="mi">12</span><span class="o">:</span><span class="mi">15</span><span class="o">:</span> <span class="n">mappedreads</span>  <span class="mi">34300176</span> 
<span class="n">INFO</span>  <span class="err">@</span> <span class="n">Mon</span><span class="p">,</span> <span class="mo">02</span> <span class="n">Feb</span> <span class="mi">2015</span> <span class="mi">08</span><span class="o">:</span><span class="mi">12</span><span class="o">:</span><span class="mi">15</span><span class="o">:</span> <span class="n">zerosgrnas</span>   <span class="mi">119315</span> 
<span class="n">INFO</span>  <span class="err">@</span> <span class="n">Mon</span><span class="p">,</span> <span class="mo">02</span> <span class="n">Feb</span> <span class="mi">2015</span> <span class="mi">08</span><span class="o">:</span><span class="mi">12</span><span class="o">:</span><span class="mi">15</span><span class="o">:</span> <span class="n">label</span>        <span class="n">sample_1</span> 
<span class="n">INFO</span>  <span class="err">@</span> <span class="n">Mon</span><span class="p">,</span> <span class="mo">02</span> <span class="n">Feb</span> <span class="mi">2015</span> <span class="mi">08</span><span class="o">:</span><span class="mi">12</span><span class="o">:</span><span class="mi">15</span><span class="o">:</span> <span class="n">Summary</span> <span class="n">of</span> <span class="n">file</span> <span class="n">sample2_R1</span><span class="p">.</span><span class="n">fastq</span><span class="o">:</span> 
<span class="n">INFO</span>  <span class="err">@</span> <span class="n">Mon</span><span class="p">,</span> <span class="mo">02</span> <span class="n">Feb</span> <span class="mi">2015</span> <span class="mi">08</span><span class="o">:</span><span class="mi">12</span><span class="o">:</span><span class="mi">15</span><span class="o">:</span> <span class="n">reads</span>        <span class="mi">36344414</span> 
<span class="n">INFO</span>  <span class="err">@</span> <span class="n">Mon</span><span class="p">,</span> <span class="mo">02</span> <span class="n">Feb</span> <span class="mi">2015</span> <span class="mi">08</span><span class="o">:</span><span class="mi">12</span><span class="o">:</span><span class="mi">15</span><span class="o">:</span> <span class="n">mappedreads</span>  <span class="mi">27042629</span> 
<span class="n">INFO</span>  <span class="err">@</span> <span class="n">Mon</span><span class="p">,</span> <span class="mo">02</span> <span class="n">Feb</span> <span class="mi">2015</span> <span class="mi">08</span><span class="o">:</span><span class="mi">12</span><span class="o">:</span><span class="mi">15</span><span class="o">:</span> <span class="n">zerosgrnas</span>   <span class="mi">119002</span> 
<span class="n">INFO</span>  <span class="err">@</span> <span class="n">Mon</span><span class="p">,</span> <span class="mo">02</span> <span class="n">Feb</span> <span class="mi">2015</span> <span class="mi">08</span><span class="o">:</span><span class="mi">12</span><span class="o">:</span><span class="mi">15</span><span class="o">:</span> <span class="n">label</span>        <span class="n">sample_2</span>
</pre></div>
<p>It provides the total number of reads, the number of mapped reads, the number of sgRNAs with 0 read counts, and the sample label of the fastq file.</p>
<h2 id="installation-problems">Installation problems</h2>
<p><strong>Q</strong>: I encountered an error after installation: "ImportError: No module named mageck". What is the problem?<br>
<strong>A</strong>: Probably you are installing MAGeCK to your own directory, which is not recognized by Python. The solution is to set up the PYTHONPATH environment: see <a class="" href="https://sourceforge.net/p/mageck/wiki/install/#setting-up-the-environment-variables">install/#setting-up-the-environment-variables</a> for more details.</p></div>
</div>
<div>
<div class="markdown_content"><h1 id="version-history">Version history</h1>
<div class="toc">
<ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#version-history">Version history</a><ul>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#version-042">Version 0.4.2</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#041">0.4.1</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#04">0.4</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#03">0.3</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#02">0.2</a></li>
<li><a href="https://sourceforge.net/p/mageck/wiki/Home/#01">0.1</a></li>
</ul>
</li>
</ul>
</div>
<h2 id="version-042">Version 0.4.2</h2>
<p>2015.02.04 Version 0.4.2</p>
<ul>
<li>Create youtube tutorial videos for installation and sample comparisons.</li>
<li>Improve the median normalization method to handle cases with many zero-count sgRNAs.</li>
<li>The median normalized read count are provided in the count command.</li>
<li>Modify the count command line options to accept combining reads from technical replicates.</li>
<li>Provide simple statistics for processing fastq files.</li>
<li>Provide library file for Synergistic Activation Mediators (SAM), a CRISPR activation protocol developed in Feng Zhang laboratory (http://www.addgene.org/crispr/libraries/sam/).</li>
</ul>
<h2 id="041">0.4.1</h2>
<p>2014.12.01 Version 0.4.1</p>
<ul>
<li>Increase the default alpha cutoff from 0.05 to 0.25.</li>
<li>Provide some of the commonly used library files for the convenience of users.</li>
</ul>
<h2 id="04">0.4</h2>
<p>2014.11.13 Version 0.4</p>
<ul>
<li>Added the BSD license information.</li>
<li>Improved the logging system.</li>
<li>The control_id and treatment_id options now can be specified using sample strings.</li>
<li>Merge positive selection and negative selection genes and pathways into one file.</li>
<li>Add the --keep-tmp option to control intermediate files after running.</li>
<li>Fixed one bug in FDR calculation.</li>
</ul>
<h2 id="03">0.3</h2>
<p>2014.07.01 Version 0.3</p>
<ul>
<li>The installation method is changed so users can now more easily install the software.</li>
<li>Added a new feature to detect enriched pathways (pathway command)</li>
<li>Changed the input format of the program:<ul>
<li>The second column of the count table (generated by the count subcommand and used by the test subcommand) is now the gene name.</li>
<li>For the count subcommand, the sgRNA information is provided with the library file.</li>
</ul>
</li>
</ul>
<h2 id="02">0.2</h2>
<p>2014.04.17 Version 0.2</p>
<ul>
<li>Updated the demo and wiki page</li>
</ul>
<h2 id="01">0.1</h2>
<p>2014.04.04 Version 0.1</p>
<ul>
<li>The source code released.</li>
</ul></div>
</div>
<h6>Project Members:</h6>
<ul class="md-users-list">
<li><a href="https://sourceforge.net/u/davidliwei">Wei Li</a> (admin)</li>
</ul>
</div></div></div>

  <div id="create_wiki_page_holder" title="Create New Page" style="display:none">
    <form>
        <label class="grid-2">Name</label>
        <div class="grid-7"><input type="text" name="name" tabindex="3"></div>
    </form>
  </div>

          </div>
			
          
  <hr style="margin-top: 1em; margin-bottom: 2em; clear:both;">
  
  
    <h4>Related</h4>
    <p>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/QA/">Wiki: QA</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/demo/">Wiki: demo</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/history/">Wiki: history</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/input/">Wiki: input</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/install/">Wiki: install</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/libraries/">Wiki: libraries</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/output/">Wiki: output</a><br>
    
      
        
        
          
        
      
      <a href="https://sourceforge.net/p/mageck/wiki/usage/">Wiki: usage</a><br>
    
    </p>
  
  
    
    <div>

<div>
                <div class="row">
                        <div class="column grid_12">
      
        <div>
  <div class="page_list">
    
  </div>
  <div class="clear"></div>
</div>
      
      <div id="comment">
        
          
        
          <div id="allow_moderate"></div>
        
      </div>
      <div style="clear:both"></div>
      
      <div style="clear:both"></div>
    </div>
  </div>
</div>

  <div id="new_post_holder">
    <span id="add_comment"></span>
    <div class="row reply">
      <div class="grid-2 tcenter">
          <p class="gravatar">
            
  
    <img src="./MAGeCK_Wiki_Home_files/b9cc1f9048ed9e437d36915feecd38c5" alt="Wei Li" title="Wei Li" class="emboss x48">
  <br>
            <small>
               
        
            <a href="https://sourceforge.net/u/davidliwei/">
                
                    Wei Li
                
            </a>
        
    
            </small>
          </p>
        </div>
        <div class="grid-17">
        
<div>
  <form method="post" action="https://sourceforge.net/p/mageck/wiki/_discuss/thread/8575cbc1/post" enctype="multipart/form-data">
    
    
<div class="markdown_edit">
  <a href="https://sourceforge.net/p/mageck/wiki/Home/#" class="markdown_preview btn" title="Preview"><b data-icon="s" class="ico ico-search"></b> Preview</a>
  <a href="https://sourceforge.net/p/mageck/wiki/Home/#" class="markdown_edit btn" style="display:none" title="Edit"><b data-icon="p" class="ico ico-pencil"></b> Edit</a>
  <a href="https://sourceforge.net/p/mageck/wiki/markdown_syntax_dialog" class="markdown_help btn" title="Formatting Help"><b data-icon="h" class="ico ico-help"></b> Formatting Help</a>
  <div style="clear:both"></div>
  <textarea id="w-2f.text" name="XXlV0GfYini2DwZH5hAz9oZnJN-4" class="auto_resize" style="height:7em; width:97%" tabindex="4"></textarea>
  <div class="markdown_preview" style="display:none"></div>
  <a href="https://sourceforge.net/p/mageck/wiki/Home/#" class="markdown_preview btn" title="Preview"><b data-icon="s" class="ico ico-search"></b> Preview</a>
  <a href="https://sourceforge.net/p/mageck/wiki/Home/#" class="markdown_edit btn" style="display:none" title="Edit"><b data-icon="p" class="ico ico-pencil"></b> Edit</a>
  <div class="modal markdown_help" style="display:none">
    <b data-icon="D" class="ico ico-close close"></b>
    <div class="markdown_help_contents"></div>
  </div>
  <input type="hidden" class="markdown_project" value="mageck" tabindex="5">
  <input type="hidden" class="markdown_neighborhood" value="4be2faf8898e33156f00003d" tabindex="6">
  <input type="hidden" class="markdown_app" value="wiki" tabindex="7">
</div>
    <input type="hidden" class="original_value" value="" tabindex="8">
    <span class="arw"><span></span></span><br>
    <div style="clear:both"></div>
    <input type="submit" value="Post" tabindex="9">
    <a href="https://sourceforge.net/p/mageck/wiki/Home/#" class="btn link cancel_edit_post">Cancel</a> &nbsp;
    <a href="https://sourceforge.net/p/mageck/wiki/Home/#" class="btn link attachment_form_add_button">Add attachments</a>
    <input type="file" class="text attachment_form_fields" style="display:none" multiple="True" name="file_info" tabindex="10">
  
  <input id="w-30" name="timestamp" type="hidden" value="1423234349" tabindex="11">
  <input id="w-31" name="spinner" type="hidden" value="XWiERYYKmWlTk6hzkf1WFf_B_hOI" tabindex="12">
  <p class="XRnlGCMfAwvhuIKzzFoUtM">
    <label for="XXUl-D-ffamRgLmWDVNiYhKkHWos">You seem to have CSS turned off.
        Please don't fill out this field.</label><br>
    <input id="XXUl-D-ffamRgLmWDVNiYhKkHWos" name="XXEl-D-ffatAgk3vP8kh-Joih7VQ" type="text" tabindex="13"><br></p>
  <p class="XRnlGCMfAwvhuIKzzFoUtM">
    <label for="XXUl-D-ffamVgLmWDVNiYhKkHWos">You seem to have CSS turned off.
        Please don't fill out this field.</label><br>
    <input id="XXUl-D-ffamVgLmWDVNiYhKkHWos" name="XXEl-D-ffa9Agk3vP8kh-Joih7VQ" type="text" tabindex="14"><br></p>
    
    <input name="_session_id" type="hidden" value="2a1b4b1bf30258e98895b2920cd6cb87e6b9aab4aa794c4c6bd05ab1ea23ba582268f7d5b93f92a4" tabindex="15">
  
  </form>
</div>
      </div>
    </div>
  </div>
</div>
  
  <div id="lightbox_confirm" class="modal" style="display:none">
  <b data-icon="D" class="ico ico-close close"></b>
  
    
  
</div>


        </div>
      
      </div>
    </section>
      
<footer id="site-footer">
    <div class="wrapper">
        <nav>
            <h5>SourceForge</h5>
            <a href="https://sourceforge.net/about">About</a>
            <a href="https://sourceforge.net/blog/category/sitestatus/">Site Status</a>
            <a href="http://twitter.com/sfnet_ops">@sfnet_ops</a>
            <a id="allura-notice" href="https://forge-allura.apache.org/p/allura/">Powered by Allura</a>
        </nav>
        <nav>
            <h5>Find and Develop Software</h5>
            <a href="https://sourceforge.net/create/">Create a Project</a>
            <a href="https://sourceforge.net/directory/">Software Directory</a>
            <a href="https://sourceforge.net/top">Top Downloaded Projects</a>
        </nav>
        <nav>
            <h5>Community</h5>
            <a href="https://sourceforge.net/blog/">Blog</a>
            <a href="http://twitter.com/sourceforge">@sourceforge</a>
            <a href="https://sourceforge.net/jobs?source=footer">Job Board</a>
            <a href="http://library.slashdotmedia.com/?source=sfnet_footer">Resources</a>
        </nav>
        <nav>
            <h5>Help</h5>
            <a href="http://p.sf.net/sourceforge/docs">Site Documentation</a>
            <a href="https://sourceforge.net/support">Support Request</a>
            <a href="http://p.sf.net/sourceforge/irc">Real-Time Support</a>
        </nav>
    </div>
</footer>
<footer id="site-copyright-footer">
    <div class="wrapper">
        <div id="copyright">
            Copyright Â© 2015 SourceForge. All Rights Reserved.<br>
            SourceForge is a <a href="http://www.diceholdingsinc.com/phoenix.zhtml?c=211152&p=irol-landing">Dice Holdings, Inc.</a> company.
        </div>
        <nav>
            <a href="http://slashdotmedia.com/terms-of-use">Terms</a>
            <a href="http://slashdotmedia.com/privacy-statement/">Privacy</a>
            <span id="teconsent"></span>
            <a href="http://slashdotmedia.com/opt-out-choices">Opt Out Choices</a>
            <a href="http://slashdotmedia.com/">Advertise</a>
            <a href="http://sourceforge.jp/">SourceForge.JP</a>
        </nav>
    </div>
</footer>
    <div id="messages" style="position: fixed; top: 20px;">
        
    </div>
    
    
      <!-- ew:body_js -->

    
      <script type="text/javascript" src="./MAGeCK_Wiki_Home_files/js"></script>
    
      
<!-- /ew:body_js -->

    
    
      <!-- ew:body_js_tail -->

    
      <script type="text/javascript">$(function () {
            var $lightbox = $('#lightbox_confirm');
            var $trigger = $('a.post-link');
            $trigger.bind('click', function(e) {
                $lightbox.lightbox_me({ modalCSS: { minHeight: 0, width: 'inherit', top: '150px'}});
                return false;
            });
            $($lightbox).delegate('.close', 'click', function(e) {
                $lightbox.trigger('close');
                return false;
            });
        });
$(function () {
            var $lightbox = $('#lightbox_create_wiki_page');
            var $trigger = $('#sidebar a.add_wiki_page');
            $trigger.bind('click', function(e) {
                $lightbox.lightbox_me();
                return false;
            });
            $($lightbox).delegate('.close', 'click', function(e) {
                $lightbox.trigger('close');
                return false;
            });
        });
$(function () {
            $('#lightbox_create_wiki_page form').submit(function(){
                location.href = $('#sidebar a.add_wiki_page').attr('href') +
                    encodeURIComponent($('input[name=name]', $(this)).val().replace('/', '-')) + '/edit';
                return false;
            });
        });
$(function () {
            $('textarea.auto_resize').focus(function(){$(this).autosize();});
        });
$(document).ready(function () {
            $("a.attachment_form_add_button").click(function(evt){
                $(this).hide();
                $(".attachment_form_fields", this.parentNode).show();
                evt.preventDefault();
            });
            $("a.cancel_edit_post").click(function(evt){
                $("textarea", this.parentNode).val($("input.original_value", this.parentNode).val());
                $(".attachment_form_fields input", this.parentNode).val('');
                evt.preventDefault();
            });
         });

        (function () {
            $('div.discussion-post').each(function () {
                var post = this;
                $('.submit', post).button();
                $('.flag_post', post).click(function (ele) {
                    this.parentNode.submit();
                    return false;
                });
                $('.moderate_post', post).click(function(e){
                    e.preventDefault();
                    var mod = $(this).text();
                    if (mod === 'Delete' && !confirm('Really delete this post?')) {
                        return;
                    }
                    var id_post = $(post).attr('id');
                    $.ajax({
                        type: 'POST',
                        url: this.parentNode.action,
                        data: jQuery(this.parentNode).serialize(),
                        success: function() {
                            if (mod == 'Delete'){
                                $(post).remove();
                            }
                            else if (mod == 'Approve'){
                                $('a.reply_post, a.shortlink, form.moderate_spam, form.moderate_approve', post).toggle();
                                $('div.moderate', post).removeClass('moderate');
                            }
                            else if (mod == 'Spam'){
                                $(post).remove();
                            }
                        }
                    });
                });

                if($('a.edit_post', post)){
                    $('a.edit_post', post).click(function (ele) {
                        $('.display_post', post).hide();
                        $('.edit_post_form', post).show();
                        // Calling jQuery's ".focus()" forces browser to reload page
                        // while using IE11 with sourceforge theme
                        $('.edit_post_form textarea', post)[0].focus();
                        return false;
                    });
                    $("a.cancel_edit_post", post).click(function(evt){
                        $('.display_post', post).show();
                        $('.edit_post_form', post).hide();
                    });
                }
                if($('.reply_post', post)){
                    $('.reply_post', post).click(function (ele) {
                        $('.reply_post_form', post).show();
                        // Calling jQuery's ".focus()" forces browser to reload page
                        // while using IE11 with sourceforge theme
                        $('.reply_post_form textarea', post)[0].focus();
                        return false;
                    });
                    $('.reply_post', post).button();
                }
                if($('.add_attachment', post)){
                    $('.add_attachment', post).click(function (ele) {
                        $('.add_attachment_form', post).show();
                        return false;
                    });
                }
                if($('.shortlink', post)){
                    var popup = $('.shortlink_popup', post);
                    $('.shortlink', post).click(function(evt){
                        evt.preventDefault();
                        popup.lightbox_me({
                            onLoad: function() {
                                $('input', popup).select();
                            }
                        });
                    });
                    $('.close', popup).bind('click', function() {
                        popup.hide();
                    });
                }
            });
        }());
        
$(function () {
            $('select.results_per_page').change(function () {
                this.form.submit();});});
$(function () {
          $('input.label_edit').tagsInput({
              'height':'100%',
              'width':'100%',
              'autocomplete_url':'/p/mageck/wiki/tags'
          });
          $('form').on('blur', '.ui-autocomplete-input', function() {
              setTimeout(function(){
                  var clicked = $(document.activeElement); // This is the element that has focus
                  if (clicked.is('#ui-active-menuitem')) {
                      return false;
                  } else {
                      var value = $('div.tagsinput div input').val();
                      var exists = $('input.label_edit').tagExist(value);
                      var default_value = $('div.tagsinput div input').attr('data-default');
                      if ((value !== default_value) && (!exists) && value !== '') {
                          $('input.label_edit').addTag(value);
                      }
                      $('input[type=submit]', this).attr('disabled', 'disabled');
                  }
                }, 1);
            });
        });

        $(document).ready(function () {
            var thread_tag = $('a.thread_tag');
            var thread_spam = $('a.sidebar_thread_spam');
            var tag_thread_holder = $('#tag_thread_holder');
            var allow_moderate = $('#allow_moderate');
            var mod_thread_link = $('#mod_thread_link');
            var mod_thread_form = $('#mod_thread_form');
            if (mod_thread_link.length) {
                if (mod_thread_form.length) {
                    mod_thread_link.click(function (e) {
                        mod_thread_form.show();
                        return false;
                    });
                }
            }
            if (thread_tag.length) {
                if (tag_thread_holder.length) {
                    var submit_button = $('input[type="submit"]', tag_thread_holder);
                    var cancel_button = $('<a href="#" class="btn link">Cancel</a>').click(function(evt){
                        evt.preventDefault();
                        tag_thread_holder.hide();
                        thread_tag.removeClass('active');
                    });
                    submit_button.after(cancel_button);
                    thread_tag.click(function (e) {
                        tag_thread_holder.show();
                        thread_tag.addClass('active');
                        // focus the submit to scroll to the form, then focus the subject for them to start typing
                        submit_button.focus();
                        $('input[type="text"]', tag_thread_holder).focus();
                        return false;
                    });
                }
            }
            if (thread_spam.length) {
                if (allow_moderate.length) {
                    var cval = $.cookie('_session_id');
                    thread_spam[0].style.display='block';
                }
            }
        });
        

            $(function(){$(".download-button-533ec35981b24b2aced2594a").load("/projects/mageck/best_release.html");
        });</script>
    
      
<!-- /ew:body_js_tail -->

    
    
  <div id="lightbox_create_wiki_page" class="modal" style="display:none">
  <b data-icon="D" class="ico ico-close close"></b>
  
    
    <h1>Add a Wiki Page</h1>
    <form class="grid-10">
        <label class="grid-2">Name</label>
        <div class="grid-7"><input type="text" name="name" tabindex="16"></div>
        <label class="grid-2">&nbsp;</label>
        <div class="grid-7"><input type="submit" value="Create page" tabindex="17"></div>
    </form>
  
  
</div>
    <script type="text/javascript">
        $('.post-link').click(function () {
            var version = $(this).data("dialog-id");
            if (version) {
                var dialog_text = $(".confirmation_dialog_" + version);
            } else {
                var dialog_text = $(".confirmation_dialog");
            }

            var modal = $('#lightbox_confirm');
            modal.html(dialog_text.html());

            modal.find('.continue_confirm').click(function (evt) {
                var cval = $.cookie('_session_id');
                evt.preventDefault();
                $.post($(this).attr('href'), {_session_id: cval}, function (val) {
                    window.location = val.location;
                }, 'json');
            });
        });
    </script>
  

    
      
    
    
   
    <script src="./MAGeCK_Wiki_Home_files/webtracker.js" type="text/javascript"></script>
    <!-- Google Code for Remarketing tag -->
    <!-- Remarketing tags may not be associated with personally identifiable information or placed on pages related to sensitive categories. For instructions on adding this tag and more information on the above requirements, read the setup guide: google.com/ads/remarketingsetup -->
    <script type="text/javascript">
        /* <![CDATA[ */
        var google_conversion_id = 1002083962;
        var google_conversion_label = "G_uGCOaBlAQQ-qzq3QM";
        var google_custom_params = window.google_tag_params;
        var google_remarketing_only = true;
        /* ]]> */
    </script>
    <script type="text/javascript" src="./MAGeCK_Wiki_Home_files/conversion.js"> </script><iframe name="google_conversion_frame" title="Google conversion frame" width="0" height="0" src="about:blank" frameborder="0" marginwidth="0" marginheight="0" vspace="0" hspace="0" allowtransparency="true" scrolling="no" style="display: none !important; visibility: hidden !important; opacity: 0 !important; background-position: 300px 13px;">&lt;img height="1" width="1" border="0" alt="" src="https://googleads.g.doubleclick.net/pagead/viewthroughconversion/1002083962/?frame=0&amp;random=1423234361724&amp;cv=7&amp;fst=1423234361724&amp;num=1&amp;fmt=1&amp;label=G_uGCOaBlAQQ-qzq3QM&amp;guid=ON&amp;u_h=900&amp;u_w=1440&amp;u_ah=805&amp;u_aw=1440&amp;u_cd=24&amp;u_his=33&amp;u_tz=-300&amp;u_java=true&amp;u_nplug=18&amp;u_nmime=50&amp;frm=0&amp;url=https%3A//sourceforge.net/p/mageck/wiki/Home/&amp;ref=https%3A//sourceforge.net/p/mageck/wiki/history/&amp;vis=1" /&gt;</iframe>
    <script type="text/javascript" src="./MAGeCK_Wiki_Home_files/get" id="truste_0.7779271858744323"></script>
    <noscript>
      &lt;div style="display:inline;"&gt;
        &lt;img height="1" width="1" style="border-style:none;" alt="" src="//googleads.g.doubleclick.net/pagead/viewthroughconversion/1002083962/?value=0&amp;amp;label=G_uGCOaBlAQQ-qzq3QM&amp;amp;guid=ON&amp;amp;script=0"/&gt;
      &lt;/div&gt;
    </noscript>

     
      
    
  
<script id="hiddenlpsubmitdiv" style="display: none;"></script><script>try{for(var lastpass_iter=0; lastpass_iter < document.forms.length; lastpass_iter++){ var lastpass_f = document.forms[lastpass_iter]; if(typeof(lastpass_f.lpsubmitorig2)=="undefined"){ lastpass_f.lpsubmitorig2 = lastpass_f.submit; lastpass_f.submit = function(){ var form=this; var customEvent = document.createEvent("Event"); customEvent.initEvent("lpCustomEvent", true, true); var d = document.getElementById("hiddenlpsubmitdiv"); if (d) {for(var i = 0; i < document.forms.length; i++){ if(document.forms[i]==form){ if (typeof(d.innerText) != 'undefined') { d.innerText=i; } else { d.textContent=i; } } } d.dispatchEvent(customEvent); }form.lpsubmitorig2(); } } }}catch(e){}</script></body></html>